py/desispec/zcatalog.py

```diff
@@ -1,15 +1,429 @@
+    
+    columns_list : list 
+        If all_columns = False, list of columns to include in the final table.
+        If None, a list of pre-decided summary columns will be used. Default is None.
+        The columns_list must include 'TARGETID', 'ZWARN', and 'TSNR2_LRG' columns.
+
+    Returns
+    -------
+    
+    table : Astropy table
+        Final table with added empty columns as needed.
+
+    """
+    
+    ## Spectral Directory Path for a given internal release name
+    specred_dir = f'/global/cfs/cdirs/desi/spectro/redux/{spec_release}'
+    
+    ## Directory path to all the redshift catalogs
+    zcat_dir = f'{specred_dir}/zcatalog'
+    
+    ## Get the filename of the redshift catalog given its survey and program
+    if (version == 'zpix'):
+        # zpix-survey-program.fits
+        filename = f'{zcat_dir}/zpix-{survey}-{program}.fits'
+    elif (version == 'ztile'):
+        # Considering only *cumulative.fits files: ztile-survey-program-cumulative.fits
+        filename = f'{zcat_dir}/ztile-{survey}-{program}-cumulative.fits'
+    else:
+        # If version = something else by mistake
+        # The error message will be printed and the function will exit.
+        print ('WARNING: version not recognized')
+        return
+
+    ## Load the ZCATALOG table, along with the meta data
+    tab = Table.read(filename, hdu = 'ZCATALOG')
+    
+    ## Removing the SURVEY and PROGRAM metadata
+    ## This is because we are stacking catalogs from multiple surveys and programs
+    del tab.meta['SURVEY']
+    del tab.meta['PROGRAM']
+    ## We keep the rest of the meta data 
+    
+    ## Add the SURVEY and PROGRAM columns
+    ## SURVEY is added as a str7 and PROGRAM is added as str6 (to match other catalogs) 
+    ## 'special' consists of seven characters and is the maximum character for a SURVEY string
+    ## 'backup' and 'bright' consists of six characters and is the maximum for a PROGRAM string
+    col1 = Column(np.array([survey]*len(tab)), name = 'SURVEY', dtype = '<U7')
+    col2 = Column(np.array([program]*len(tab)), name = 'PROGRAM', dtype = '<U6')
+    tab.add_column(col1, 1)
+    tab.add_column(col2, 2)
+    ## The SURVEY and PROGRAM columns are added as second and third columns,
+    ## immediately after TARGETID
+    
+    ## Add the TARGET columns -- only for SV, not needed for Main
+    if (spec_release == 'fuji'):
+        ## Add the other target columns that are not present in the redshift catalogs
+        ## Depends on the survey and program
+        ## The function `add_target_columns` adds the required columns based on survey and program
+        tab = add_target_columns(tab, survey = survey, program = program)
+        
+        ## Making a list of all the '*_TARGET* columns 
+        ## This is for rearranging the columns into proper order
+        target_cols = ['CMX_TARGET', 'DESI_TARGET','BGS_TARGET','MWS_TARGET', 'SCND_TARGET',\
```

Comment:
These backslashes are completely unnecessary and my trigger style warnings in the future.  In general, a line continuation backslash is never needed in any modern version of Python.